lightningd/subd.c

```diff
@@ -749,8 +749,14 @@ void subd_req_(const tal_t *ctx,
 	add_req(ctx, sd, type, num_fds_in, replycb, replycb_data);
 }
 
+static void discard_alarm(int sig UNNEEDED)
```

Comment:
So we are discarding the alarm itself, we are simply depending on signals causing syscalls to return `EINTR`?  That seems quite clever.  Maybe a comment or two about this?